subiquity/server/controllers/filesystem.py

```diff
@@ -491,6 +495,16 @@ async def guided(self, choice: GuidedChoiceV2):
         if gap is None:
             raise Exception('failed to locate gap after adding boot')
 
+        if choice.reset_partition:
+            cp = await arun_command(['du', '-sb', '/cdrom'])
+            reset_size = int(cp.stdout.strip().split()[0])
+            reset_size = align_up(reset_size + GiB, GiB)
+            reset_gap, gap = gap.split(reset_size)
+            self.reset_partition = self.create_partition(
+                device=reset_gap.device, gap=reset_gap,
+                spec={'fstype': 'fat32'})
```

Comment:
if the symlinks become a problem I guess we could dd the ISO directly to the partiton.

Reply:
Well this depends on details I don't entirely understand, but under some circumstances we want to be able to boot directly from the reset partition (i.e. the firmware loads shim and then grub from it) in which case it has to be fat32 and not iso9660.